src/Cli/dotnet/ToolManifest/ToolManifestFinder.cs

```diff
@@ -162,12 +164,48 @@ public FilePath FindFirst()
                     return possibleManifest;
                 }
             }
-
+            if (createManifestFileOption)
+            {
+                DirectoryPath manifestInsertFolder = GetInsertManifestFilePath();
+                if (manifestInsertFolder.Value != null)
+                {
+                    return new FilePath(WriteManifestFile(manifestInsertFolder));
+                }
+            }
             throw new ToolManifestCannotBeFoundException(
-                LocalizableStrings.CannotFindAManifestFile,
-                string.Format(LocalizableStrings.ListOfSearched,
-                    string.Join(Environment.NewLine,
-                        EnumerateDefaultAllPossibleManifests().Select(f => "\t" + f.manifestfile.Value))));
+                    LocalizableStrings.CannotFindAManifestFile,
+                    string.Format(LocalizableStrings.ListOfSearched,
+                        string.Join(Environment.NewLine,
+                            EnumerateDefaultAllPossibleManifests().Select(f => "\t" + f.manifestfile.Value))));
+        }
+
+        private DirectoryPath GetInsertManifestFilePath()
```

Comment:
I'm not sure what a good name for this method would be, but I think this is a bit of an improvement:

```suggestion
        private DirectoryPath GetDirectoryToCreateToolManifest()
```